### PR TITLE
New indexed field inserted_at_num in blocktx db

### PR DESCRIPTION
**Pull request by boecklim**
Adding an index on a timestamp of insertion is needed for eventual deletion of the records. Otherwise deletions will be very slow when a lot of records exist. An index on a timestamp turns out to have a penalty on performance, which can be seen when e2e tests fail in this case. Thus a new field was added which represents the hour when a record was added.

- New indexed field inserted_at_num in blocktx db which represents the hour when a record was added as an integer
- Update integration tests with new indexed field
- Change log entry `Status not updated for tx` to debug log level

### CODE REVIEW

File: background_worker/jobs/clear_block_transactions_map.go [hunk 1]
```diff
@@ -9,23 +9,24 @@ import (
...
 	conn, err := sqlx.Open(params.Scheme(), params.String())
 	if err != nil {
 		Log(ERROR, fmt.Sprintf("unable to create connection %s", err))
 		return err
 	}
-	interval := fmt.Sprintf("%d days", params.RecordRetentionDays)
 
-	stmt, err := conn.Preparex("DELETE FROM block_transactions_map WHERE inserted_at <= (CURRENT_DATE - $1::interval)")
+	start := c.now()
+	datenumRetentiondDays := start.Add(-24 * time.Hour * time.Duration(params.RecordRetentionDays))
```
Comment (nozim) [inline background_worker/jobs/clear_block_transactions_map.go:21]: RecordRetentionDays is already in Days so this parameter seems redundant to me.

File: background_worker/jobs/clear_block_transactions_map_test.go [hunk 2]
```diff
@@ -24,36 +25,30 @@ func (s *ClearBlockTransactionsMapSuite) Test() {
...
+		testfixtures.Directory("fixtures"), // The directory containing the YAML files
+	)
+	require.NoError(s.T(), err)
 
-	db, err := sqlx.Open("postgres", params.String())
+	err = fixtures.Load()
 	require.NoError(s.T(), err)
 
-	err = ClearBlockTransactionsMap(params)
+	now := time.Date(2023, 12, 22, 12, 0, 0, 0, time.UTC)
```
Comment (nozim) [inline background_worker/jobs/clear_block_transactions_map_test.go:41]: you don't need to manipulate date inside ClearJob. 
You simply insert "old" records into db and make sure it's deleted after job is complete.

File: background_worker/jobs/clear_blocks.go [hunk 3]
```diff
@@ -10,29 +10,56 @@ import (
...
 		Log(ERROR, "unable to create connection")
 		return err
 	}
-	interval := fmt.Sprintf("%d days", params.RecordRetentionDays)
 
-	stmt, err := conn.Preparex("DELETE FROM blocks WHERE inserted_at <= (CURRENT_DATE - $1::interval)")
+	start := c.now()
+	datenumRetentiondDays := start.Add(-24 * time.Hour * time.Duration(params.RecordRetentionDays))
+
+	stmt, err := conn.Preparex("DELETE FROM blocks WHERE inserted_at_num <= $1::int")
```
Comment (nozim) [inline background_worker/jobs/clear_blocks.go:56]: what is inserted_at_num and how does it differ from inserted_at? Why we can't use inserted_at?

File: database/migrations/blocktx/postgres/000005_add_inserted_at.up.sql [hunk 4]
```diff
@@ -1,5 +1,12 @@
+ALTER TABLE blocks
+ADD COLUMN inserted_at_num INTEGER DEFAULT TO_NUMBER(TO_CHAR((NOW()) AT TIME ZONE 'UTC', 'yyyymmddhh24'), '9999999999') NOT NULL;
```
Comment (nozim) [inline database/migrations/blocktx/postgres/000005_add_inserted_at.up.sql:2]: Why do we need this column?

**Comment by shotasilagadzetaal** [comment]
Probably too late but we could add block num there and remove blocks below specific height.

File: database/migrations/blocktx/postgres/000005_add_inserted_at.up.sql (hunk 4 above)
Comment (shotasilagadzetaal) [reply database/migrations/blocktx/postgres/000005_add_inserted_at.up.sql:2]: Will this default value correctly convert to hours?

**Review by shotasilagadzetaal — APPROVED** [review]
We could also use timestamp (by changing the format to unix time, same as integer and index it) and not add a new column

File: database/migrations/blocktx/postgres/000005_add_inserted_at.up.sql (hunk 4 above)
Comment (boecklim) [reply database/migrations/blocktx/postgres/000005_add_inserted_at.up.sql:2]: In my test it converted to hours. When I do this for example:
`SELECT TO_NUMBER(TO_CHAR((NOW()) AT TIME ZONE 'UTC', 'yyyymmddhh24'), '9999999999');`
I get
`2023121112`

**Pull request merged**